File: RKGeonames/AdministrationData.m
//
//  AdministrationData.m
//  RKGeonames
//
//  Created by Stefan Burettea on 13/09/2013.
//  Copyright (c) 2013 Stefan Burettea. All rights reserved.
//

#import "AdministrationData.h"

@implementation AdministrationData


// |+|=======================================================================|+|
// |+|                                                                       |+|
// |+|    FUNCTION NAME: initWithCapitalCity                                 |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    DESCRIPTION:                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    PARAMETERS:                                                        |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    RETURN VALUE:  N/A                                                 |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|=======================================================================|+|
- (id)initWithCapitalCity:(NSString *)capitalCity
                  surface:(NSString *)surface
              currentTime:(NSString *)currentTime
                 timeZone:(NSString *)timeZone
                  sunrise:(NSString *)sunrise
                   sunset:(NSString *)sunset
{
    self = [super init];
    if(self)
    {
        _capitalCity = capitalCity;
        _surface = surface;
        _currentTime = currentTime;
        _timeZone = timeZone;
        _sunrise = sunrise;
        _sunset = sunset;
    }
    
    return self;
}

+ (id)data;
{
    return [[self alloc] initWithCapitalCity:@"N/A"
                                   surface:@"N/A"
                               currentTime:@"N/A"
                                  timeZone:@"N/A"
                                   sunrise:@"N/A"
                                    sunset:@"N/A"];

}

@end

File: CountriesOfTheWorld/AdministrationData.swift
//
//  AdministrationData.swift
//  RKGeonames
//
//  Created by Stefan Buretea on 9/20/14.
//  Copyright (c) 2014 Stefan Burettea. All rights reserved.
//

import Foundation

class AdministrationData : NSObject {
    
    let capitalCity: NSString
    let surface: NSString
    let currentTime: NSString
    let timeZone: NSString
    let sunrise: NSString
    let sunset: NSString
    
    init(capitalCity: NSString, surface: NSString, currentTime: NSString, timeZone: NSString, sunrise: NSString, sunset: NSString) {
        
        self.capitalCity = capitalCity
        self.surface = surface
        self.currentTime = currentTime
        self.timeZone = timeZone
        self.sunrise = sunrise
        self.sunset = sunset
    }
    
    class func data() -> AdministrationData {
        
        return AdministrationData(capitalCity: "" as NSString!,
                                  surface: "" as NSString!,
                                    currentTime: "" as NSString!,
                                    timeZone: "" as NSString!,
                                    sunrise: "" as NSString!,
                                    sunset: "" as NSString!)
    }
    
}